src/python/module/smartredis/configoptions.py

```diff
@@ -0,0 +1,225 @@
+# BSD 2-Clause License
+#
+# Copyright (c) 2021-2023, Hewlett Packard Enterprise
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from .smartredisPy import PyConfigOptions
+from .util import exception_handler, typecheck
+
+from .error import RedisRuntimeError
+
+_notfactory = "Method called on a ConfigOptions object not created from a factory method"
+
+class ConfigOptions:
+    def __init__(self):
+        """Initialize a ConfigOptions base object
+        """
+        self._is_created_via_factory = False
+        self._config_opts = None
+
+    @staticmethod
+    def from_pybind(configoptions):
+        """Initialize a ConfigOptions object from
+        a PyConfigOptions object
+
+        :param configoptions: The pybind PyConfigOptions object
+                              to use for construction
+        :type dataset: PyConfigOptions
+        :return: The newly constructed ConfigOptions from
+                 the PyConfigOptions
+        :rtype: ConfigOptions
+        """
+        typecheck(configoptions, "configoptions", PyConfigOptions)
+        new_configoptions = ConfigOptions()
```

Comment:
I know we are intentionally not making `from_pybind` a classmethod so we can do a group refactor with `Dataset`, but did we ever put a ticket in for this? I'd just like to know that it is in the backlog somewhere...

Reply:
@MattToast  to write up ticket on this